quickwit-doc-mapper/src/default_doc_mapper/mod.rs

```diff
@@ -18,28 +18,32 @@
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 mod default_mapper;
+mod default_mapper_builder;
 mod field_mapping_entry;
 mod field_mapping_type;
+mod mapping_tree;
 
 use anyhow::bail;
 use once_cell::sync::Lazy;
 use regex::Regex;
 
-pub use self::default_mapper::{DefaultDocMapper, DefaultDocMapperBuilder, SortByConfig};
-pub use self::field_mapping_entry::{DocParsingError, FieldMappingEntry};
+pub use self::default_mapper::{DefaultDocMapper, SortByConfig};
+pub use self::default_mapper_builder::{DefaultDocMapperBuilder, ModeType};
+pub use self::field_mapping_entry::{FieldMappingEntry, QuickwitJsonOptions};
 pub use self::field_mapping_type::FieldMappingType;
 
 /// Regular expression validating a field mapping name.
-pub const FIELD_MAPPING_NAME_PATTERN: &str = r#"^[_a-zA-Z][_\.\-a-zA-Z0-9]{0,254}$"#;
+pub const FIELD_MAPPING_NAME_PATTERN: &str = r#"^[a-zA-Z][_\.\-a-zA-Z0-9]{0,254}$"#;
 
 /// Validates a field mapping name.
 /// Returns `Ok(())` if the name can be used for a field mapping. Does not check for reserved field
 /// mapping names such as `_source`.
 ///
 /// A field mapping name:
 /// - may only contain uppercase and lowercase ASCII letters `[a-zA-Z]`, digits `[0-9]`, hyphens
-///   `-`, periods `.`, and underscores `_`;
-/// - must start with an uppercase or lowercase ASCII letter `[a-zA-Z]`, or an underscore `_`;
+///   `-`, and underscores `_`;
```

Comment:
I'm ok to ban field names that start with `_`. I'm expecting to see users with fields starting `_` to be happy with json field.